graph/graph_structure.js

```diff
@@ -0,0 +1,174 @@
+// First, let us begin with a M x N matrix
+// Each element of the matrix is a node and we can
+
+class Graph {
+
+    constructor(m=3, n=3) {
+        this.columns = m;
+        this.rows = n;
+        this.visited = [];
+        this.graph = new Map();
+        this.goal=-1;
+    }
+    set_goal(goal){
+        this.goal = goal;
+    }
+    get_graph() {
+        console.log(this.graph);
+        return this.graph;
+    }
```

Comment:
ES6 has a slick getter/setter syntax if youre interested.

The result is that code using the getter/setter just looks like you're working with an attribute (i.e. `this.val = 1` could run setter code if `this.val` was a defined as a setter; and `this.val` could run getter code if `this.val` was implemented as a getter.

idk if its worth refactoring, but its nice to know about for the case of originally implementing something as just an attribute and then realizing some get/set logic would be nice (cause then you wouldn't have to change any code using the old attribute).

Reply:
Definitely worth looking into.